File: Data Structures & Algorithms/two-integer-sum/submission-1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {}
        for i, num in enumerate(nums):
            hash[num] = i
        
        for i, num in enumerate(nums):
            if target-num in hash and i != hash[target-num]:
                return [i, hash[target-num]]
        return [0,0]

File: Data Structures & Algorithms/house-robber/submission-1.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n = nums.size();
        if(n < 3)
            return n == 1 ? nums[0] : max(nums[0], nums[1]);
            
        vector<int> money(n, 0);
        money[0] = nums[0]; money[1] = max(nums[0], nums[1]);
        for(int i=2; i<n; i++)
            money[i] = max(money[i-1], money[i-2] + nums[i]);

        return money[n-1];
    }
};